Client/Frontend/SiriShortcuts.swift

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Intents
+import IntentsUI
+import Shared
+
+@available(iOS 12.0, *)
+class SiriShortcuts {
+    enum activityType: String {
+        case openURL = "org.mozilla.ios.Firefox.newTab"
+    }
+
+    func getActivity(for type: activityType) -> NSUserActivity? {
+        switch type {
+        case .openURL:
+            return openUrlActivity
+        }
+    }
+
+    private var openUrlActivity: NSUserActivity? = {
+        let activity = NSUserActivity(activityType: activityType.openURL.rawValue)
+        activity.title = Strings.SettingsSiriOpenURL
+        activity.userInfo = ["action": "newTab"]
```

Comment:
If you switch on activityType in AppDelegate, userInfo should become unnecessary.